src/couch/src/couch_doc.erl

```diff
@@ -275,9 +275,16 @@ transfer_fields([{<<"_revisions">>, {Props}} | Rest], Doc, DbName) ->
     true ->
         ok
     end,
-    [throw({doc_validation, "RevId isn't a string"}) ||
-            RevId <- RevIds, not is_binary(RevId)],
-    RevIds2 = [parse_revid(RevId) || RevId <- RevIds],
+    RevIds2 = lists:map(fun(RevId) ->
+        try
```

Comment:
any reason the check isn't  in `parse_revid` given the other `parse_revs` also throw errors?

Reply:
> given the other parse_revs also throw errors?
what do you mean? as far as I see `parse_revid/1` doesn't have `throw` in any clauses.

Reply:
@eiri : I should be a little specific. I just meant other `parse_rev` functions such as 

https://github.com/apache/couchdb/blob/57c0f04c681cd69806a525801a9431a4f9161cde/src/couch/src/couch_doc.erl#L185

and 

https://github.com/apache/couchdb/blob/57c0f04c681cd69806a525801a9431a4f9161cde/src/couch/src/couch_doc.erl#L198

were throwing errors.

Reply:
@tonysun83 hm, ok. Well it is technically possible to convert `parse_revid/1` to do the same, but that'd require to add some kind of validation step there or introduce re-throw block to raise proper error type and will spread `doc_validation` exception cases across the module instead of keeping them within `transfer_fields/3` as it is now. In my opinion these two are pretty much equal approaches to fix this, so why do you think your idea is better than what I have here?

Reply:
@eiri: My nitpick is only for readability purposes. Like you said, if we move it to `parse_revid/1`, you'd have to add a validation step and re-throw. The advantage of that is that you'd know we're throwing it because of `erlang:list_to_integer/2` and can logically conclude that it's not a valid hex. Whereas inside `transfer_fields/3`, and first glance, you can't assume there's only one possibility for `bad_args` and have to read `parse_revid`. Having said that, you do make a good point of keeping it inside `transfer_fields/3` so that it doesn't permeate other parts of the codebase, especially since `parse_rev/1` calls `parse_revid`. We don't want a change in behavior if 400 errors thrown suddenly in unexpected places (even if it is correct). I'll leave it up to you if you want to add a comment to make it a little clearer or if you think it should be self-evident as it is.

Reply:
there are two possibilities for `badarg`, actually, an inconvertible list to `?l2b` is the second, but honestly, I think we (server side) don't care _why_ exactly parse failed as long as we can pass back a message that the rev format is invalid and then let the user to figure it out. this is a simple fix for a simple bug and I don't feel that overcomplicating things with validators and whatnot will add any value here.